libs/utils/android/screen.py

```diff
@@ -155,4 +157,57 @@ def unlock(target):
        System.menu(target)
        System.home(target)
 
+    @staticmethod
+    def record_start(target, out_file,
+                     timestamp=True, max_time=None):
+        """
+        Start recording the screen of the device
+        """
+        log = logging.getLogger('Screen')
+
+        options = []
+
+        # Populate recording options
+        if timestamp:
+            options.append('--bugreport')
+        if max_time:
+            options.append('--time-limit {}'.format(max_time))
+
+        remote_file = target.path.join(target.working_directory,
+                                       os.path.basename(out_file))
+
+        record_cmd = 'screenrecord {} {}'.format(
+            ' '.join(options), remote_file
+        )
+
+        target.adb_root(enable=True, force=True)
+
+        log.debug('screenrecord command = \"{}\"'.format(record_cmd))
+        log.info('Starting screen recording')
+
+        return {
+            'command' : target.background(record_cmd),
+            'remote_file': remote_file,
+            'local_file': out_file,
+            'max_time': max_time
+        }
+
+    @staticmethod
+    def record_stop(target, screenrecord_data):
+        """
+        Stop recording the screen of the device
+        """
+        log = logging.getLogger('Screen')
+
+        if not screenrecord_data['max_time']:
+            # Recording must be ended by Ctrl-C on the device
+            target.execute('kill -INT $(pgrep screenrecord)')
```

Comment:
Ah yeah, you're right. If you want to keep the target abstraction you can probably use `target.get_pids_of()` before `target.kill()` though. However, my previous tests with `get_pids_of()` were not very successful (`ps` is not always behaving as expected on Android)  and, if this is really Android-specific, it's probably OK to leave it as it is